embed/serve.go

```diff
@@ -101,6 +101,7 @@ func (sctx *serveCtx) serve(
 		opts := []grpc.DialOption{grpc.WithInsecure()}
 		gwmux, err := sctx.registerGateway(opts)
 		if err != nil {
+			gs.Stop()
```

Comment:
probably make error as named return. and we always try to stop gs if it is not nil

Reply:
Done.